src/move/move.ts

```diff
@@ -77,11 +78,34 @@ export const checkUserMove = (
     };
   }
 
-  if (playedTile.symbol === NEUTRALE_TILE.symbol) {
+  const lastPlayedTile = getLastPlayedTile(board);
+
+  const isNeutraleTile = playedTile.symbol === NEUTRALE_TILE.symbol;
+  const isColorConstraintUnrespected =
+    playedTile.color !== lastPlayedTile.color;
+  const isSymbolConstraintUnrespected =
+    playedTile.symbol !== lastPlayedTile.symbol;
+
+  if (
+    isNeutraleTile ||
+    (lastPlayedTile != null &&
+      isColorConstraintUnrespected &&
+      isSymbolConstraintUnrespected)
+  ) {
+    let badMoveMessage = `🫠 Tile is not playable. Please player ${gameState.currentPlayer.toUpperCase()} choose a playable tile`;
+
+    if (isColorConstraintUnrespected) {
+      badMoveMessage = `🫠 Tile has a different COLOR. Please player ${gameState.currentPlayer.toUpperCase()} choose a playable tile`;
+    }
+
+    if (isSymbolConstraintUnrespected) {
+      badMoveMessage = `🫠 Tile has a different SYMBOL. Please player ${gameState.currentPlayer.toUpperCase()} choose a playable tile`;
+    }
```

Comment:
This method is already too long =>  All this new code should be moved to a new function

Reply:
I refactored it.